src/query/api/v1/options/handler.go

```diff
@@ -175,6 +188,20 @@ type HandlerOptions interface {
 	DefaultQueryEngine() QueryEngine
 	// SetDefaultQueryEngine returns the default query engine.
 	SetDefaultQueryEngine(value QueryEngine) HandlerOptions
+
+	// QueryRouter is a reference to the router which is responsible for routing
+	// queries between PromQL and M3Query.
+	// Optional. If not set it will default to default query router.
```

Comment:
nit: no need to flag it as optional, most of these are optional with sane defaults ;)